### PR TITLE
Add personal token support

### PR DESCRIPTION
**Pull request by gjsduarte**
Adds personal token / app password support for pushing tags via https

### CODE REVIEW

**Comment by andreaTP** [comment]
This PR is breaking, do you mind to change the commit message to have this reflected into the version number?

File: src/jobs/tag_version.yml [hunk 1]
```diff
@@ -6,20 +6,34 @@ parameters:
...
+    description: "Access token/app password of the account used to push to Bitbucket/Github"
+    type: string
+    default: "$GPR_AUTH_TOKEN"
 
 executor: machine
 
 steps:
   - attach_workspace:
       at: ~/workdir
-  - add_ssh_keys:
```
Comment (andreaTP) [inline src/jobs/tag_version.yml:18]: keeping those lines will make this PR not breaking, I assume.

File: src/jobs/tag_version.yml [hunk 2]
```diff
@@ -6,20 +6,34 @@ parameters:
     type: string
     default: "$(cat .version)"
   force:
-    description: "Use force to override versions"
+    description: "Use the force to override versions"
```
Comment (andreaTP) [inline src/jobs/tag_version.yml:9]: Brilliant!

File: src/jobs/tag_version.yml (hunk 1 above)
Comment (gjsduarte) [reply src/jobs/tag_version.yml:18]: It isn't technically breaking as I already added the environment variables to CircleCI. But I agree with marking the commit as breaking.

**Comment by gjsduarte** [comment]
> This PR is breaking, do you mind to change the commit message to have this reflected into the version number?

It isn't technically breaking as I already added the environment variables to CircleCI. But I agree with marking the commit as breaking.

File: src/jobs/tag_version.yml (hunk 1 above)
Comment (rtfpessoa) [reply src/jobs/tag_version.yml:18]: How is this going to work for bitbucket repositories?

File: src/jobs/tag_version.yml [hunk 3]
```diff
@@ -6,20 +6,34 @@ parameters:
...
     type: boolean
     default: false
+  username:
+    description: "Username of the account used to push to Bitbucket/Github"
+    type: string
+    default: "codacycircleci"
+  password:
+    description: "Access token/app password of the account used to push to Bitbucket/Github"
+    type: string
+    default: "$GPR_AUTH_TOKEN"
```
Comment (rtfpessoa) [inline src/jobs/tag_version.yml:19]: What does `GPR` stand for?

File: src/jobs/tag_version.yml (hunk 1 above)
Comment (rtfpessoa) [reply src/jobs/tag_version.yml:18]: Answer: Since circleci has different contexts per organization

File: src/jobs/tag_version.yml (hunk 3 above)
Comment (rtfpessoa) [reply src/jobs/tag_version.yml:19]: Answer: GitPRovider_AUTH_TOKEN

**Pull request merged**